lib/selenium-cucumber/click_elements_steps.rb

```diff
@@ -1,9 +1,12 @@
 require_relative 'methods/click_elements_methods'
 
-# click on web element
-When(/^I click on element having (.+) "(.*?)"$/) do |type, access_name|
+When(/^I click on element having (id|name|class|xpath|css) "(.*?)"(?: and text "(.*?)")?$/) do |type, access_name, text|
```

Comment:
well, (.+) will pretty much aggressively match anything, by specifying exactly which locator_types are matched seems more safe.  Also, the (.+) because of its aggressive matching, doesn't separate the arguments (specifically if the optional is provided).  I like being more explicit here since selenium in the end only has N locators.